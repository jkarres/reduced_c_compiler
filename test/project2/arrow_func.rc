structdef LIST {
  int i;
  LIST* next;

  function : void print() {
    cout << this.i;

    if (this.next != NULL) {
      cout << " ";
      this.next->print();
    }
  }

};

LIST[10] theList;

function : void main() {
  int i;

  for (i = 0; ; ++i) {
    theList[i].i = i*10;

    if (i == 9) {
      break;
    }

    theList[i].next = &theList[i+1];
  }

  cout << "printing out each element directly:" << endl;
  for (i = 0; i < 10; ++i) {
    cout << theList[i].i << endl;
  }

  cout << "printing out using the print function:" << endl;
  theList[0].print();
  cout << endl;

}

