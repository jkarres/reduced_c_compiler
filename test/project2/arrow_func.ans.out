printing out each element directly:
0
10
20
30
40
50
60
70
80
90
printing out using the print function:
0 10 20 30 40 50 60 70 80 90
